projects/plugins/jetpack/modules/google-fonts.php

```diff
@@ -115,6 +132,63 @@ function jetpack_add_google_fonts_provider() {
 }
 add_action( 'after_setup_theme', 'jetpack_add_google_fonts_provider' );
 
+/**
+ * Updates font names by filtering the data provided by the theme for global styles & settings.
+ *
+ * `wp_register_webfonts()` does not currently support giving a separate "name"
+ * from "font-family", even if theme.json schema itself does support it.
+ * See: https://github.com/WordPress/gutenberg/issues/46398
+ *
+ * Once support for "name" is added, below functionality can be refactored
+ * to provide the name directly within `wp_register_webfonts()`.
+ *
+ * @param WP_Theme_JSON_Data_Gutenberg $theme_json Class to access and update the underlying data.
+ * @return WP_Theme_JSON_Data_Gutenberg Class with updated Global Styles settings.
+ */
+function jetpack_rename_google_font_names( $theme_json ) {
+	$raw_data      = $theme_json->get_data();
+	$font_families = $raw_data['settings']['typography']['fontFamilies']['theme'];
```

Comment:
Do all block themes have all of these keys defined? If not it might make sense to return early if an `isset()` returns false.

Reply:
Will `JETPACK_GOOGLE_FONTS_LIST` be part of `$font_families`? Or is `$font_families` the static representation of that theme json data?

Reply:
I added a few checks;
- tests if array keys exist
- Skip renaming entirely for non wp-admin/rest API requests.